cli/src/config.rs

```diff
@@ -326,11 +329,31 @@ impl UnresolvedConfigEnv {
 pub struct ConfigEnv {
     home_dir: Option<PathBuf>,
     repo_path: Option<PathBuf>,
+    workspace_root: Option<PathBuf>,
     user_config_paths: Vec<ConfigPath>,
     repo_config_path: Option<ConfigPath>,
     command: Option<String>,
 }
 
+pub struct RepoManagedConfigPaths {
+    /// The path to the file managed by the repo.
+    pub managed: PathBuf,
```

Comment:
I feel like this should be a `RepoPathBuf` and that the contents should be read from the working-copy commit. That's generally the same as reading it from the working copy; the main difference is that it works even if the config file is sparsed away. I'm guess the feature should ideally work even if the user has configured the sparse patterns to match only `lib/`, for example. Even if you agree, I don't think you need to implement it right now. Feel free to leave a TODO IMO.